src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -563,6 +576,22 @@ async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryA
             return await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(instanceId);
         }
 
+        async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryAsync(IEnumerable<string> instanceIds)
+        {
+            int instanceDeleted = 0;
+            var tasks = instanceIds.Select(async instanceId =>
+            {
+                var purgeHistoryResult = await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(instanceId);
+                if (purgeHistoryResult.InstancesDeleted == 1)
+                {
+                    instanceDeleted += 1;
+                }
+            });
+            await Task.WhenAll(tasks);
+            PurgeHistoryResult result = new PurgeHistoryResult(instancesDeleted: instanceDeleted);
+            return result;
```

Comment:
As with my previous comment, I recommend this rewrite:

```suggestion
            var tasks = instanceIds.Select(instanceId => this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(instanceId));
            var results = await Task.WhenAll(tasks);

            var instancesDeleted = 0;
            foreach (var purgeHistoryResult in results)
            {
                if (purgeHistoryResult.InstancesDeleted == 1)
                {
                    instancesDeleted += 1;
                }
            }

            PurgeHistoryResult result = new PurgeHistoryResult(instancesDeleted: instancesDeleted);
            return result;
```

There's 2 main changes here:
(1) We're constructing the list of tasks directly, without the need to declare local async functions
(2) We're not modifying the shared `instancesDeleted` variable inside the Tasks, which is preferrable for data integrity in multi-threaded code :)